src/arrayMethodSort.js

```diff
@@ -4,8 +4,22 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  const defaultCompare = function defaultCompare(a, b) {
+    return a.toString() > b.toString();
+  };
+
+  [].__proto__.sort2 = function (compareFunction = defaultCompare) {
+    const compare = compareFunction;
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compare(this[j], this[j + 1]) > 0) {
```

Comment:
The comparison function is expected to return a number, but here it is implicitly cast to a boolean. Ensure that the comparison function is used correctly by comparing its return value to 0 instead of casting it to a boolean.